Source/WebCore/animation/ScrollTimeline.cpp

```diff
@@ -138,4 +140,39 @@ AnimationTimeline::ShouldUpdateAnimationsAndSendEvents ScrollTimeline::documentW
     return AnimationTimeline::ShouldUpdateAnimationsAndSendEvents::No;
 }
 
+ScrollableArea* ScrollTimeline::scrollableAreaForSourceRenderer(CheckedPtr<RenderBox> renderer, Ref<Document> document)
+{
+    return renderer->element() == document->documentElement() ? (ScrollableArea*)&renderer->view().frameView() : (ScrollableArea*)renderer->layer()->scrollableArea();
```

Comment:
I don't think you need either of the `(ScrollableArea*)` casts.

Reply:
Also you are not checking `renderer->hasLayer()`, so `renderer->layer()` will often crash. Don't you need to find the renderer's enclosing ScrollableArea somewhere?

Reply:
Right added a check for layer() and canBeScrolledAndHasScrollableArea(). We don't need to look for the enclosing scrollable area, that is just for ViewTimelines. ScrollTimelines are provided the source element when created through the api.